sui/token_bridge/sources/register_chain.move

```diff
@@ -55,11 +55,10 @@ module token_bridge::register_chain {
     }
 
     public entry fun submit_vaa(wormhole_state: &mut WormholeState, bridge_state: &mut BridgeState, vaa: vector<u8>, ctx: &mut TxContext) {
-        let vaa = token_bridge_vaa::parse_verify_and_replay_protect(wormhole_state, bridge_state, vaa, ctx);
+        let vaa = corevaa::parse_and_verify(wormhole_state, vaa, ctx);
```

Comment:
why change from `parse_verify_and_replay_protect`?

Reply:
it was broken up into `parse_and_verify` and then `replay_protect`. One has to call `corevaa::parse_and_verify` instead of `parse_and_verify`, because the latter checks if the vaa chain is registered, so a vaa for registering a chain would never pass the check

Reply:
ah right, I see that's how we did in aptos too